### PR TITLE
Improve name resolution in Liquid Haskell

### PR DESCRIPTION
**Pull request by facundominguez**

### CODE REVIEW

**Review by yannham — COMMENTED** [review]
I think the description does a good job at explaining the technical context. Is there any size limit? If there's a bit more room, it might make the proposal more attractive to include even just one sentence about the motivation for something like LiquidHaskell (something about ensuring high assurance/correctness for critical software in a non-obtrusive way, for example).

Also, I wonder why name resolution is currently done two times? Is it for ease of implementation, for historical reasons, or because of objective technical difficulties that the candidate will have to solve?

File: content/ideas/liquidhaskell-name-resolution.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* What names should be accessible in the specifications of a given module?
+
+* What should be the mechanism to bring names in scope?
+
+* How should the system negotiate or resolve ambiguous names?
+
+* Which design and/or implementation steps should be done to adjust Liquid
+  Haskell to the proposed mechanisms?
+
+* How much effort would the previous steps take?
```
Comment (smelc) [inline content/ideas/liquidhaskell-name-resolution.md]: I find it a bit surprising to require a proposal. It looks like there is some pre-project acceptance criteria here (which I did not in see in [existing proposals](https://github.com/haskell-org/summer-of-haskell/tree/master/content/ideas)), whereas I think there isn't any. You are simply outlaying what the project will look like right? If that's the case, I would reword to make this look less intimidating.

Comment (facundominguez) [reply content/ideas/liquidhaskell-name-resolution.md]: I reduced the intimidation factor. But since GSoC already provides the guidelines for proposals, maybe I should just keep the requirement of the final blog post.

**Comment by facundominguez** [comment]
> Is there any size limit?

Only the patience of the student :) I'll consider your suggestion. 

> Also, I wonder why name resolution is currently done two times?

I don't know for sure, but my guess is it is a historical reason. Meaning that different projects modified the code base without consolidated goals.

I don't anticipate technical obstacles that justify the current state of affairs, though I admit perhaps we learn something more about it during the project.

File: content/ideas/liquidhaskell-name-resolution.md (hunk 1 above)
Comment (facundominguez) [reply content/ideas/liquidhaskell-name-resolution.md]: > I should just keep the requirement of the final blog post.

Done.

File: content/ideas/liquidhaskell-name-resolution.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+a blog post summarizing the achievements.
+
+As a stretch goal, this project might specify and fix the scope rules
+and mechanisms to use in Liquid Haskell specifications.
+While the intention of Liquid Haskell maintainers over time has been to
+imitate GHC scoping rules as much as possible, there are cases where Liquid
+Haskell just deviates from them.
+
+The following aspects haven't been answered explicitly so far for the Liquid
+Haskell scenario.
```
Comment (smelc) [inline content/ideas/liquidhaskell-name-resolution.md]:
```suggestion
In other words, to attain this supplementary stretch goal, one will need to answer those questions:
```

Rationale: I think that's what you mean

Comment (facundominguez) [reply content/ideas/liquidhaskell-name-resolution.md]: Thanks @smelc! Adopted with edits.

**Comment by facundominguez** [comment]
@aaronallen8455, I think this is good to merge unless you have suggestions.
